bionemo-recipes/models/llama3/create_tokenizer.py

```diff
@@ -0,0 +1,228 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+
+"""
+Script to create the HuggingFace PreTrainedTokenizerFast for nucleotide sequences.
+
+This script creates a tokenizer that:
+1. Maps each character to its ord() value (ASCII encoding)
+2. Uses special tokens with NeMo convention (EOS=0, PAD=1, BOS=2, UNK=3)
+3. Works with AutoTokenizer.from_pretrained()
+
+Run this script to regenerate the tokenizer files if needed.
+"""
+
+import logging
+import os
+import tempfile
+
+import torch
+from tokenizers import Tokenizer, processors
+from tokenizers.models import WordLevel
+from tokenizers.pre_tokenizers import Split
+from transformers import AutoTokenizer, DataCollatorForLanguageModeling, PreTrainedTokenizerFast
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def create_nucleotide_tokenizer(
+    eos_id: int = 0,
+    pad_id: int = 1,
+    bos_id: int = None,
+    unk_id: int = 2,
+) -> PreTrainedTokenizerFast:
+    """
+    Create a PreTrainedTokenizerFast for nucleotide sequences.
+    
+    Follows NeMo ByteTokenizer convention:
+    - eos_id = 0
+    - pad_id = 1  
+    - bos_id = None (optional, for causal LM we'll use 2)
+    
+    Args:
+        eos_id: End-of-sequence token ID (NeMo convention: 0)
+        pad_id: Padding token ID (NeMo convention: 1)
+        bos_id: Beginning-of-sequence token ID (None in NeMo, but we use 2 for causal LM)
+        unk_id: Unknown token ID (2 or 3)
+        
+    Returns:
+        PreTrainedTokenizerFast ready to use and save
+    """
+    # Define special tokens with NeMo convention
+    special_tokens = {
+        "<EOS>": eos_id,
+        "<PAD>": pad_id,
+        "<UNK>": unk_id,
+    }
+    
+    if bos_id is not None:
+        special_tokens["<BOS>"] = bos_id
+    
+    # Build vocab: Map each ASCII character to its ord() value
+    # IMPORTANT: Exclude chr(0-3) to reserve those IDs for special tokens
+    vocab = {**special_tokens}
+    reserved_ids = {eos_id, pad_id, unk_id}
+    if bos_id is not None:
+        reserved_ids.add(bos_id)
```

Comment:
another nit, but why duplicate this logic? `set(special_tokens.values())` would work, right?